src/main/java/com/google/api/generator/engine/ast/CastExpr.java

```diff
@@ -52,17 +52,18 @@ public CastExpr build() {
       // Only numeric types can be casted around.
       if (castType.isPrimitiveType() && exprType.isPrimitiveType()) {
         // Check that the types are either both booleans or both not booleans.
-        // TODO(xiaozhenliu): This is one of the places where we need to check the equality of boxed
-        // and primitive types.
         Preconditions.checkState(
             !(castType.equals(TypeNode.BOOLEAN) ^ exprType.equals(TypeNode.BOOLEAN)),
             "Numeric and boolean types are not inter-castable");
       } else {
+        boolean isExprTypeReferenceOrNull =
```

Comment:
This looks like the CastExpr PR, can we keep this delta out of this PR? Same for the other changes.

Reply:
Sure, done.